riscv/mmu.cc

```diff
@@ -299,7 +299,7 @@ reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, access_type type, access_type trap_ty
         break;
       } else if (!proc->extension_enabled(EXT_SVNAPOT) && (pte & PTE_N)) {
         break;
-      } else if (!proc->extension_enabled(EXT_SVPBMT) && (pte & PTE_PBMT)) {
+      } else if (!(proc->get_state()->csrmap[CSR_HENVCFG]->read() & HENVCFG_PBMTE) && (pte & PTE_PBMT)) {
```

Comment:
Prefer accessing the CSR object directly, i.e. `proc->get_state()->henvcfg->read()`. The `csrmap` is there for dynamic access such as `csrw` instruction execution where the specific CSR is unknown at Spike compile time.